zenodo/modules/theme/templates/zenodo_theme/security/login_user.html

```diff
@@ -26,8 +26,8 @@
 
 {%- block form_outer %}
   <div class="social-signup">
-    <a href="{{url_for('invenio_oauthclient.login', remote_app='github')}}" class="btn btn-default btn-lg btn-block"><i class="fa fa-github"></i> {{_('Log in with GitHub')}}</a>
-    <a href="{{url_for('invenio_oauthclient.login', remote_app='orcid')}}" class="btn btn-default btn-lg btn-block"><img src="{{ url_for('static', filename='img/orcid.svg')}}" height="18px"> {{_('Log in with ORCID')}}</a>
+    <a href="{{url_for('invenio_oauthclient.login', remote_app='github', next=(request.args.get('next') or request.referrer))}}" class="btn btn-default btn-lg btn-block"><i class="fa fa-github"></i> {{_('Log in with GitHub')}}</a>
```

Comment:
I think we will need to sanitize the next URL to include only local URLs. We have a utility function somewhere to do this, just don't remember where right now (accounts, oauthclient or oauth2server I believe).

Reply:
I assume [invenio_oauthclient.utils.get_safe_redirect_target](https://github.com/inveniosoftware/invenio-oauthclient/blob/2359e19caa9ef1063a4b16f0d1dc648755c5823a/invenio_oauthclient/utils.py#L175-L184), which is already used inside the [`/oauth/login/<remote_app>`](https://github.com/inveniosoftware/invenio-oauthclient/blob/7ef6bfbcf9a4eec3ee78f6fb46edfd7495d82fb8/invenio_oauthclient/views/client.py#L89) view. Also this is kind of already protected because it's passed to the OAuth workflow which will reject it in case it's not a [sub-path of the default `redirect_uri`](https://developer.github.com/apps/building-integrations/setting-up-and-registering-oauth-apps/about-authorization-options-for-oauth-apps/#redirect-urls)

Reply:
Ah, correct! I was looking in ``authorized`` view :-( Yes, you're right it's already protected.

Regarding the GitHub callback URL that's a separate thing. The ``next`` parameter is something completely isolated to Invenio. I.e. GitHub redirects to /oauth/authorized/github/, which redirects to the next URL.